bundles/org.openhab.binding.buienradar/src/main/java/org/openhab/binding/buienradar/internal/BuienradarHandler.java

```diff
@@ -97,25 +100,51 @@ public void initialize() {
             updateStatus(ThingStatus.UNKNOWN);
         }
         if (listenableFuture == null || listenableFuture.isCancelled()) {
-            listenableFuture = scheduler.scheduleWithFixedDelay(this::refresh, 0L, config.refreshIntervalMinutes,
-                    MINUTES);
+            listenableFuture = scheduler.scheduleWithFixedDelay(() -> refresh(config), 0L,
```

Comment:
Wouldn't it be easier to store the configuration in the handler and use it when you need it?

Reply:
I am trying to keep mutable state in the handler to a minimum. Now it is easy to prove that `initialize()` will always re-read the configuration.

Reply:
I guess I'm used to an FP, immutability kind of thinking. 'config' as a field in your proposal would need to be a mutable and nullable field, making it harder to reason whether config is accessed from the right methods (from `initialize` instead of many of the other methods). This places `config` in the lambda closure, as if I had created a new `Runnable` with one field initialized in the constructor called `config`.  It prevents needless `null`-checks, possible race-conditions and reduced efferent coupling with other state.

So, yes, I considered making `config` a field, but decided against it, since it is only one field, it is the defining part of the closure (no other fields needed) and (imho) made the code more readable.

But if you disagree... Well, the code needs to be merged and it would make a large number of people happy, so I'm more than happy to have you make the suggested change.

Reply:
Correction, I am more than happy to make that change.